### DISCUSSION TITLE
Formatting Not Working

### DISCUSSION THREAD

**Discussion by tails1154** [CosmosOS/Cosmos#2958] [General]
#### Area of Cosmos - What area of Cosmos are we dealing with?
formatting drives

#### Expected Behaviour - What do you think that should happen?
the format method should work

#### Actual Behaviour - What unexpectedly happens?
cannot find the method

#### Reproduction - How did you get this error to appear?
![asasasass](https://github.com/CosmosOS/Cosmos/assets/152324947/d00182c3-48d3-412f-a814-a9b30b813b6b)

#### Version - Were you using the User Kit or Dev Kit? And what User Kit version or Dev Kit commit (Cosmos, IL2CPU, X#)?
User Kit Cosmos

**Comment by zarlo** [comment]
you need to get the disk you want to format then use [FormatPartition](https://cosmosos.github.io/api/cosmos/Cosmos.System.FileSystem.Disk.html#Cosmos_System_FileSystem_Disk_FormatPartition_System_Int32_System_String_System_Boolean_)

**Comment by tails1154** [comment]
The method does not exist

**Comment by zarlo** [comment]
[FormatPartition](https://cosmosos.github.io/api/cosmos/Cosmos.System.FileSystem.Disk.html#Cosmos_System_FileSystem_Disk_FormatPartition_System_Int32_System_String_System_Boolean_) is on the Disk class

**Comment by tails1154** [comment]
I am looking there

**Comment by zarlo** [comment]
are you new to C# Disk is not static

**Comment by tails1154** [comment]
![image](https://github.com/CosmosOS/Cosmos/assets/152324947/423cb002-2d00-4693-98fe-b004103ac9fb)
yes

**Comment by tails1154** [comment]
i found it

**Comment by zarlo** [comment]
it has some static method bot most of them are not

**Comment by tails1154** [comment]
![image](https://github.com/CosmosOS/Cosmos/assets/152324947/cf40ccf1-b84c-4bff-8877-3695bce6a1a5)

**Comment by tails1154** [comment]
how do I find my blockchain?

**Comment by zarlo** [comment]
there will be a list of disks the in VFS

**Comment by tails1154** [comment]
is it like 0:\ ?

**Comment by tails1154** [comment]
but how do I convert my text to a blockchain?

**Comment by zarlo** [comment]
what text?

**Comment by tails1154** [comment]
0:\

**Comment by tails1154** [comment]
can you please just give me a command that sets up the blockchain thing?

**Comment by zarlo** [comment]
what blockchain thing?

**Comment by tails1154** [comment]
I have to go, but
this blockchain thing:
![Screenshot 2024-02-20 213934](https://github.com/CosmosOS/Cosmos/assets/152324947/4c6a5303-ddf8-44a5-acd2-713466f1d821)

**Comment by zarlo** [comment]
you get it from the disk list in the VFS

**Comment by tails1154** [comment]
how would I print that to the string because Object.ToString(); not yet implemented?

**Comment by tails1154** [comment]
sorry console

**Comment by blackneeed** [comment]
Please switch to devkit its better

**Comment by tails1154** [comment]
fInE

**Comment by tails1154** [comment]
i finally installed it

**Comment by tails1154** [comment]
oh gosh this post is old
